templates/kubernetes/terraform/modules/kubernetes/aws_auth.tf

```diff
@@ -0,0 +1,124 @@
+locals {
+    # Standardize the kubernetes role name
+    k8s_role_name_format = "%s-kubernetes-%s-%s" # <project_name>-kubernetes-<role_name>-<environment>
+
+    # Roles for each defined user, plus an admin user
+    comfigmap_roles = concat(
+    # Always create this admin user, as we use it by default in some of the scripts
+    [{
+      rolearn  = "arn:aws:iam::${var.allowed_account_ids[0]}:role/${format(local.k8s_role_name_format, var.project_name, "admin", var.environment)}"
+      username = format(local.k8s_role_name_format, var.project_name, "admin", var.environment)
+      groups   = ["system:masters"]
+    }],
+    [
+      for r in var.k8s_role_mapping : {
+        rolearn  = "arn:aws:iam::${var.allowed_account_ids[0]}:role/${format(local.k8s_role_name_format, var.project_name, r.name, var.environment)}"
+        username = format(local.k8s_role_name_format, var.project_name, r.name, var.environment)
+        groups   = r.k8s_groups
+      }
+    ]
+  )
+
+  # Role required by eks node group worker instances
+  worker_roles = [
+    for node_group in aws_eks_node_group.cluster :
+    {
+      rolearn  = node_group.node_role_arn
+      username = "system:node:{{EC2PrivateDNSName}}"
+      groups = [
+          "system:bootstrappers",
+          "system:nodes",
+        ]
+    }
+  ]
+}
+
+# Look up the cluster node group to get the worker roles
+data "aws_eks_node_groups" "cluster" {
+  cluster_name = var.cluster_name
+}
+
+data "aws_eks_node_group" "cluster" {
+  count = count(data.aws_eks_node_groups.cluster.names)
+  cluster_name    = var.cluster_name
+  node_group_name = data.aws_eks_node_groups.cluster.names[count.index]
+}
+
+# Create the configmap
+resource "kubernetes_config_map" "aws_auth" {
+  metadata {
+    name      = "aws-auth"
+    namespace = "kube-system"
+    labels = merge(
```

Comment:
is there supposed to be another set of labels to merge with?

Reply:
No, just left over from the other module, thanks. Removed it.